### PR TITLE
BIGTOP-3371. Unable to build Apache Ambari on RHEL8.

### PR DESCRIPTION
**Pull request by sekikn**
@guyuqi @mattAndruff @masatana Would you review if possible? In this PR, I intended:

* Specify python package name explicitly as 'python2' in spec files so that they work with CentOS 8 and Fedora 31
* Replace Ambari functions that depend on the rpm-python package, which is not provided on CentOS 8, with an implementation using the `rpm` command
* Define the `python` command as an alias of `python2` on CentOS 8 and Fedora 31, which don't provide unversioned python command by default

I confirmed that Ambari, Livy and Zeppelin (the latter of two use Python3, in contrast to Ambari) are successfully built on CentOS 7, 8 and Fedora 31 with this PR.

### CODE REVIEW

File: bigtop-packages/src/common/ambari/patch4-AMBARI-CentOS8.diff [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
++++ b/ambari-common/src/main/python/resource_management/core/providers/package/__init__.py
+@@ -359,9 +359,8 @@ class RPMBasedPackageProvider(PackageProvider):
+   INSTALLED_PACKAGE_VERSION_COMMAND = "rpm -q --queryformat '%{{version}}-%{{release}}' \"{0}\""
+ 
+   def rpm_check_package_available(self, name):
+-    import rpm # this is faster then calling 'rpm'-binary externally.
+-    ts = rpm.TransactionSet()
+-    packages = ts.dbMatch()
++    import os
++    packages = os.popen("rpm -qa --queryformat '%{name} '").read().split()
```
Comment (masatana) [inline bigtop-packages/src/common/ambari/patch4-AMBARI-CentOS8.diff:26]: `packages` variable should contain a dict-like object as this is used like [this](https://github.com/apache/ambari/blob/release-2.6.1/ambari-common/src/main/python/resource_management/core/providers/package/__init__.py#L370)

**Comment by masatana** [comment]
@sekikn Thanks for working on this! I haven't tested this patch yet, but I left one comment.
I'll test this patch on my environment later.

File: bigtop-packages/src/common/ambari/patch4-AMBARI-CentOS8.diff (hunk 1 above)
Comment (sekikn) [reply bigtop-packages/src/common/ambari/patch4-AMBARI-CentOS8.diff:26]: That's right, thanks! I fixed the succeeding for-loop.

Comment (masatana) [reply bigtop-packages/src/common/ambari/patch4-AMBARI-CentOS8.diff:26]: +1. Thanks!

**Comment by masatana** [comment]
LGTM (non-binding). I also built (`./gradlew toolchain` && `./gradlew ambari-pkg`) and installed on CentOS8 with success.
If we encounter any problems with running Ambari on CentOS8, let's fix it with another addendum patch.

**Comment by sekikn** [comment]
Thank you for the review @masatana @iwasakims, just merged.

**Pull request merged**